casbin/enforcer_cached.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+* Copyright 2020 The casbin Authors. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#pragma once
+
+#include "pch.h"
+
+#include "./enforcer_cached.h"
+using namespace std;
+
+CachedEnforcer::CachedEnforcer(CachedEnforcer& ce){
+    this->e = move(ce.e);
+    this->enableCache = ce.enableCache;
+    this->m = ce.m;
+}
+
+CachedEnforcer::CachedEnforcer(unique_ptr<Enforcer>& e) {
```

Comment:
A better way here may define a move assignment operator for `Enforcer`, so no need to write `move` manually. This can be fixed after the constructor APIs have been optimized.

Reply:
> A better way here may define a move assignment operator for `Enforcer`, so no need to write `move` manually. This can be fixed after the constructor APIs have been optimized.

OK